geotrellis-sentinelhub/src/main/scala/org/openeo/geotrellissentinelhub/CatalogApi.scala

```diff
@@ -174,21 +221,24 @@ class DefaultCatalogApi(endpoint: String) extends CatalogApi {
           .valueOr(throw _)
       }
 
-      def getFeatures(limit: Int, nextToken: Option[Int]): Map[String, geotrellis.vector.Feature[Geometry, ZonedDateTime]] = {
+
+      def getFeatures(limit: Int, nextToken: Option[Int]): Map[String, geotrellis.vector.Feature[Geometry, FeatureData]] = {
         val page = getFeatureCollectionPage(limit, nextToken)
 
+        var features = Map[String, Feature[Geometry, FeatureData]]()
         // it is assumed the returned geometries are in LatLng
-        val features = page.getAllFeatures[Feature[Geometry, Json]]
-          .mapValues(feature =>
-            feature.mapData { properties =>
-              val Some(datetime) = for {
-                properties <- properties.asObject
-                json <- properties("datetime")
-                datetime <- json.asString
-              } yield ZonedDateTime.parse(datetime, ISO_OFFSET_DATE_TIME)
-
-              datetime
-            })
+        page.features.foreach { fJson =>
+          type F = Feature[Geometry, Json]
+          fJson.as[F].foreach(feature => {
+            val key = getFeatureID(fJson)
+            val newFeature = feature.mapData { properties =>
+              val Some(dateTime) = getDateTime(properties)
+              val selfUrl = getSelfUrl(fJson)
```

Comment:
`selfUrl` can be moved out of `mapData`.

Reply:
Moved in https://github.com/Open-EO/openeo-geotrellis-extensions/pull/134